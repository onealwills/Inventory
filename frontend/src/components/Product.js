import React from "react";
import { Link } from "react-router-dom";

export default function Product(props) {
  const { product } = props;
  return (
    <tr>
      <td key={product._id}>{product._id}</td>
      <td>
        <Link to={`/product/${product._id}`}>
          <img src={product.image} alt={product.name} />
        </Link>
      </td>
      <td>{product.type}</td>
      <td>{product.make}</td>
      <td>{product.model}</td>
      <td>{product.year}</td>
      <td>{product.stockQty}</td>
      <td>
        <span>&#8358;</span>
        {product.price}
      </td>
    </tr>
  );
}
