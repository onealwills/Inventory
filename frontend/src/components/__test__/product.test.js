import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Product from "../Product";

const stockKeeper = {
  _id: "2222222222222222",
  stockKeeper: { name: "john", warehouse: "Block A4, shop 12" },
};
const product = {
  _id: "1111111111111111",
  image: "/images/2009 Honda CRV drier.jpg",
  type: "Ac/drier",
  make: "Honda",
  model: "CRV",
  year: "2009",
  stockQty: 50,
  price: 12000,
  stockKeeper: stockKeeper,
};

test("<Product/> component snapshot", () => {
  const component = render(
    <MemoryRouter>
      <Product key={product._id} product={product} />
    </MemoryRouter>
  );
  expect(component.container).toMatchSnapshot();
});
