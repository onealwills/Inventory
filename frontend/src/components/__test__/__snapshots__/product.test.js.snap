// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product/> component snapshot 1`] = `
<div>
  <tr>
    <td>
      1111111111111111
    </td>
    <td>
      <a
        href="/product/1111111111111111"
      >
        <img
          src="/images/2009 Honda CRV drier.jpg"
        />
      </a>
    </td>
    <td>
      Ac/drier
    </td>
    <td>
      Honda
    </td>
    <td>
      CRV
    </td>
    <td>
      2009
    </td>
    <td>
      50
    </td>
    <td>
      <span>
        â‚¦
      </span>
      12000
    </td>
    <a
      href="/stockKeeper/2222222222222222"
    >
      <td>
        Block A4, shop 12
      </td>
    </a>
  </tr>
</div>
`;
